lib/livebook/session.ex

```diff
@@ -583,6 +600,8 @@ defmodule Livebook.Session do
              else: :ok
            ) do
       state = schedule_autosave(state)
+      Enum.each(Livebook.Settings.fetch_env_vars(), &put_env_var(self(), &1))
```

Comment:
We are already in the server process, so we can set/unset envs directly, instead of genserver-calling ourselves :)

Reply:
It only works if the Session receives the events from PubSub

Reply:
Ah, actually we don't want to do it in init, only once the runtime is started:

https://github.com/livebook-dev/livebook/blob/3e10a8fd18dfb9dd75a6ebd944c21add122dad16/lib/livebook/session.ex#L1330-L1333

And when we receive set/unset we need to check if the runtime is connected and only then apply those actions.

Reply:
Ok, now it works flawlessly